tests/irdl/test_attribute_definition.py

```diff
@@ -850,6 +803,27 @@ def test_deprecated_tuple_init():
         )
 
 
+@irdl_attr_definition
+class ParamAttrDefAttr2(ParametrizedAttribute):
+    name = "test.param_attr_def_attr"
+
+    arg1: Attribute = param_def(base(IntAttr))
+    arg2: BoolData = param_def()
```

Comment:
Pyright made me put the `param_def()` here saying that fields without default values can't follow fields that have one.